### PR TITLE
Made changes to README files to get started with Amazon Connect Chat

### PR DESCRIPTION
**Pull request by yvprashanth**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 1]
```diff
@@ -1,83 +1,137 @@
...
+Adding chat to your website is possible with a few easy steps. This solutions spins up an [Amazon API Gateway](https://aws.amazon.com/api-gateway/) endpoint that triggers an [AWS Lambda](https://aws.amazon.com/lambda/) function. This Lambda function invokes the [Amazon Connect](https://aws.amazon.com/connect/) Service [StartChatContact](https://docs.aws.amazon.com/en_pv/connect/latest/APIReference/API_StartChatContact.html) API and returns the result from that call. You could use the below pre built widget to get up and going. This solution is built on top of [Chat JS](https://github.com/amazon-connect/amazon-connect-chat-ui-examples.git). 
 
-## CloudFormation Deployment Steps
+## [Getting started with Pre Built Widget](#Getting-started-with-Pre-Built-Widget)
```
Comment (ranman) [inline cloudformationTemplates/startChatContactAPI/README.md:5]: this renders a little funny on github - might be better as a list

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 2]
```diff
@@ -1,83 +1,137 @@
...
+## [Getting started with a Custom Widget](#Getting-Started-with-a-Custom-Widget)
 
 ### Pre-requisites
```
Comment (ranman) [inline cloudformationTemplates/startChatContactAPI/README.md:8]: would make this `##` instead of `###` if you're going to make the others `##`

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 3]
```diff
@@ -1,83 +1,137 @@
...
+1. Create an Amazon Connect Instance by signing up [here](https://aws.amazon.com/connect/).
+You need an Amazon Connect instance to deploy the below [CloudFormation](https://aws.amazon.com/cloudformation/) template. You can use an existing one or create a new one by following our onboarding guide [here](https://docs.aws.amazon.com/connect/latest/adminguide/amazon-connect-get-started.html).
 
-If you are using an existing instance, you may need to make a few changes to your instance to enable Chat. Follow the steps [here](https://github.com/amazon-connect/amazon-connect-chat-ui-examples/blob/master/README.md#enabling-chat-in-an-existing-amazon-connect-contact-center) to see what changes you need to make.
+    If you are using an existing instance, you may need to make a few changes to your instance to enable Chat. Follow the steps [here](https://github.com/amazon-connect/amazon-connect-chat-ui-examples/blob/master/README.md#enabling-chat-in-an-existing-amazon-connect-contact-center) to see what changes you need to make.
 
-### Steps
+2. Once you create an Amazon Connect Instance, create a new contact flow, by picking **Sample queue customer** contact flow and **publish it**
```
Comment (ranman) [inline cloudformationTemplates/startChatContactAPI/README.md:15]: This is hard to follow. If you're going to specify what to do here you need to be very specific.

```markdown
2. After creating your instance, open the Connect console. In the navigation bar, under **Routing**, choose **Contact Flows**. Select **Sample queue customer**.
```

Why do they need to publish it? it should already be published right? Is this where you need to tell them that you want them to copy the ID and the flow ID ?

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 4]
```diff
@@ -1,83 +1,137 @@
...
-        - `instanceId`: This is the id of the instance you want to use. You can find this on the Amazon Connect console or when viewing the contact flow. For example, if the arn for your flow is 'arn:aws:connect:us-west-2:123456789012:instance/11111111-1111-1111-1111-111111111111/contact-flow/22222222-2222-2222-2222-222222222222', the instance id is '11111111-1111-1111-1111-111111111111'
-2) Once the stack has launched you can call the API from your website. Follow the steps below to see how you can call this API using our pre-built chat widget or by building out your own UX.
+3. Publish a cloud formation template using one of the regions you prefer below. 
     
-## Prebuilt Chat Widget
-If you want to add the customer chat widget (that is available in the Test Chat experience in the Amazon Connect website) to your website, here are the steps to do so. You can also refer to the `widgetIndex.html` file in this repo to see an example of how to use the widget.
+    Steps
```
Comment (ranman) [inline cloudformationTemplates/startChatContactAPI/README.md:21]: remove steps here

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 5]
```diff
@@ -1,83 +1,137 @@
...
-
-1) Deploy the CloudFormation template from one of the links above.
-    - Enter the Contact Flow id and Instance Id that you would like to test with. These are added as Lambda environment variables.
-        - `contactFlowId`: You can find the contact flow id when viewing a contact flow. For example, if the arn for your flow is 'arn:aws:connect:us-west-2:123456789012:instance/11111111-1111-1111-1111-111111111111/contact-flow/22222222-2222-2222-2222-222222222222', the contact flow id is '22222222-2222-2222-2222-222222222222'
-        - `instanceId`: This is the id of the instance you want to use. You can find this on the Amazon Connect console or when viewing the contact flow. For example, if the arn for your flow is 'arn:aws:connect:us-west-2:123456789012:instance/11111111-1111-1111-1111-111111111111/contact-flow/22222222-2222-2222-2222-222222222222', the instance id is '11111111-1111-1111-1111-111111111111'
-2) Once the stack has launched you can call the API from your website. Follow the steps below to see how you can call this API using our pre-built chat widget or by building out your own UX.
+3. Publish a cloud formation template using one of the regions you prefer below.
```
Comment (ranman) [inline cloudformationTemplates/startChatContactAPI/README.md:19]: replace publish with launch or deploy

replace `cloud formation` with `CloudFormation`

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 6]
```diff
@@ -1,83 +1,137 @@
...
+
+4. Update the CloudFormation template with details from your connect instance.
+    - Fetch the Contact ID and Instance ID from your Connect instance by going to your contact flows and selecting **Sample queue customer** and clicking on **Show additional flow information** on top left. 
+    - `contactFlowId` \ `instanceId`: You can find the contact flow id and instance id when viewing a contact flow. For example, if the arn for your flow is 'arn:aws:connect:us-west-2:123456789012:instance/11111111-1111-1111-1111-111111111111/contact-flow/22222222-2222-2222-2222-222222222222', the contact flow id is '22222222-2222-2222-2222-222222222222' and instance id is 11111111-1111-1111-1111-111111111111
+
+        ![Contact Flow Id's](images/updateContactFlowIds.png)
+
+    - Go to your [CloudFormation stack](https://us-west-2.console.aws.amazon.com/cloudformation/home) by clicking on Stacks and then startChatContactAPI on the left hand side. Update the Contact Flow ID and the Instance ID in your cloud formation stack
+    ![Update Cloud Formation Stack](images/updateCloudFormationStack.png)
+    -   These are added as Lambda environment variables.
```
Comment (ranman) [inline cloudformationTemplates/startChatContactAPI/README.md:42]: this doesn't mean much to most people following this tutorial

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 7]
```diff
@@ -1,83 +1,137 @@
...
 
-Note: you will want to show the widget only when there is a conversation in progress. If the widget is visible when there is no conversation, you will just see a loading spinner.
+## Getting started with Pre Built Widget
 
-1. In your website's html code, import the 'amazon-connect-chat-interface.js' file from this repo.
+You can refer to the `widgetIndex.html` file in this repo to see an example of how to use the widget.
 
+1. In your website's html code, import the 'amazon-connect-chat-interface.js' file from this repo. Also import jQuery
     ```html
```
Comment (ranman) [inline cloudformationTemplates/startChatContactAPI/README.md:49]: this code block is double indented.

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 8]
```diff
@@ -1,83 +1,137 @@
...
+
+    - Go to your [CloudFormation stack](https://us-west-2.console.aws.amazon.com/cloudformation/home) by clicking on Stacks and then startChatContactAPI on the left hand side. Update the Contact Flow ID and the Instance ID in your cloud formation stack
+    ![Update Cloud Formation Stack](images/updateCloudFormationStack.png)
+    -   These are added as Lambda environment variables.
 
-Note: you will want to show the widget only when there is a conversation in progress. If the widget is visible when there is no conversation, you will just see a loading spinner.
+## Getting started with Pre Built Widget
 
-1. In your website's html code, import the 'amazon-connect-chat-interface.js' file from this repo.
+You can refer to the `widgetIndex.html` file in this repo to see an example of how to use the widget.
```
Comment (ranman) [inline cloudformationTemplates/startChatContactAPI/README.md:46]: can you link to the file?

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 9]
```diff
@@ -1,83 +1,137 @@
...
-
-    ```js
-    $(document).ready((a) => {
-      connect.ChatInterface.init({
-        containerId: 'root' // This is the id of the container where you want the widget to reside
-        });
-    });
+2. Create a `div` with id of `root`. 
+    ```html
+    <div id="root">
```
Comment (ranman) [inline cloudformationTemplates/startChatContactAPI/README.md:56]: you changed this to root (it should be chat)
but you didn't change the code.

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 10]
```diff
@@ -1,83 +1,137 @@
...
+    </div>
     ```
+3. Get your API Gateway URL by going to your AWS Console and going to API Gateway product
```
Comment (ranman) [inline cloudformationTemplates/startChatContactAPI/README.md:59]: This is also an output from the CloudFormation template.

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 11]
```diff
@@ -1,83 +1,137 @@
...
+
+    ![API Gateway](images/apiGatewayUrl.png) 
 
-3. Start the chat based on a user action. You will want to add fields for the customer name and username because those fields are used in the Lambda function that was created. Note: you need to update this to include the API Gateway endpoint that was created in the CloudFormation stack. To see examples of the success and failure handlers, refer to the [example implementation](https://github.com/amazon-connect/amazon-connect-chat-ui-examples/blob/master/cloudformationTemplates/asyncCustomerChatUX/website/index.html#L283).
+4. Paste the below code as a script section in your html file and ***Update the 4 TODO fields***
 
     ```js
```
Comment (ranman) [inline cloudformationTemplates/startChatContactAPI/README.md:65]: if you're including a script tag then it should be formatted html so the syntax renders correctly.

You also double indented the script here.

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 12]
```diff
@@ -1,83 +1,137 @@
...
-      instanceId: "${instanceId}"
-    },successHandler, failureHandler)
+    <script>
+            $(document).ready((a) => {
+                connect.ChatInterface.init({
+                    containerId: 'chat' // This is the id of the container where you want the widget to reside
+                });
+            });
+
+            $(function () {
```
Comment (ranman) [inline cloudformationTemplates/startChatContactAPI/README.md:73]: Code:
why can't all of this go in the document.ready block above?
why are the variables wrapped in an if block? they could be defined outside of it so the customer would modify everything in one place?
You're asking the customer to use fields like #contactDetails but none of those exist in the customer's page you haven't told the customer how to use those or how to define them.

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 13]
```diff
@@ -1,83 +1,137 @@
...
+                        var contactFlowId = "xxxx1111-xxxx-xxxx-xxxx-xxxxxxxxxxxx"; // TODO: Fill in
+                        var instanceId = "xxxx2222-xxxx-xxxx-xxxx-xxxxxxxxxxxx"; // TODO: Fill in
+                        var apiGatewayEndpoint = "https://${apiId}.execute-api.${region}.amazonaws.com/Prod"; // TODO: Fill in with the API Gateway endpoint created by your CloudFormation template
+
+                        console.log("this is the first name: " + customerName);
+                        document.getElementById("contactDetails").reset();
+
+                        connect.ChatInterface.initiateChat({
+                            name: customerName,
+                            region: "", // TODO: Fill in
```
Comment (ranman) [inline cloudformationTemplates/startChatContactAPI/README.md:91]: this can be a variable

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 12 above)
Comment (ranman) [reply cloudformationTemplates/startChatContactAPI/README.md:73]: you've also never loaded chatui to any of these pages... does this even work?

File: README.md [hunk 14]
```diff
@@ -7,11 +7,13 @@ Before working on projects in this repo, we suggest taking time to follow [the p
 ## Solutions
 
 At the moment, these are the solutions in this repo:
-
-1) **[cloudformationTemplates/asyncCustomerChatUX](https://github.com/amazon-connect/amazon-connect-chat-ui-examples/tree/master/cloudformationTemplates/asyncCustomerChatUX)**
+ 1) **Get started with Chat** - **[CloudFormationTemplates/startChatContactAPI](https://github.com/amazon-connect/amazon-connect-chat-ui-examples/tree/master/cloudformationTemplates/startChatContactAPI)**
+    Use this solution to create a web widget with Amazon Connect Chat.
+    The Start Chat Contact API solution creates a simple API to start the chat from the customer side. Use this solution to get started with Amazon Connect Chat.
```
[the remark this answers is not in the document]
Comment (yvprashanth) [reply README.md:12]: Rephrase this to? This solution is primarily to help users get started with a web chat widget. The fact that it uses startChatAPI is underlying detail. That shouldn't be the top level message of this

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 1 above)
Comment (yvprashanth) [reply cloudformationTemplates/startChatContactAPI/README.md:5]: Ah! This was primarily to showcase at the very top that they can either choose from pre built widget or a custom widget that they can style. Can make it a bullet

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 3 above)
Comment (yvprashanth) [reply cloudformationTemplates/startChatContactAPI/README.md:15]: Are they all published to begin with? When you save something it doesn't publish them. 
Are you saying all the default contact flows are already published?

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 6 above)
Comment (yvprashanth) [reply cloudformationTemplates/startChatContactAPI/README.md:42]: Not sure why you say that?. People who aren't as familiar with AWS products may not know where to update the details for the CloudFormation stack

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 7 above)
Comment (yvprashanth) [reply cloudformationTemplates/startChatContactAPI/README.md:49]: It's rendering fine on my preview :). What do you see? 
Screen Shot 2020-04-15 at 7 09 59 PM

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 8 above)
Comment (yvprashanth) [reply cloudformationTemplates/startChatContactAPI/README.md:46]: Sure

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 9 above)
Comment (yvprashanth) [reply cloudformationTemplates/startChatContactAPI/README.md:56]: Changed it. Good catch

File: README.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (ranman) [reply README.md:12]: `Use this solution to create a web widget with Amazon Connect Chat.`
`Use this solution to get started with Amazon Connect Chat.`

Which is it? I don't think it's very clear to a reader.

Suggested rephrasing:
`Use this solution to deploy the required infrastructure to support a chat widget that can be embedded on different pages.`

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 11 above)
Comment (yvprashanth) [reply cloudformationTemplates/startChatContactAPI/README.md:65]: Not sure how you find double indentation. I see it as underneath the bullet point
Screen Shot 2020-04-15 at 7 13 12 PM

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 12 above)
Comment (yvprashanth) [reply cloudformationTemplates/startChatContactAPI/README.md:73]: I added the details of html page. It does work. Try after the new commit

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 3 above)
Comment (ranman) [reply cloudformationTemplates/startChatContactAPI/README.md:15]: yes they are all published by default.

I suggest you run through your own instructions on a fresh connect instance.

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 7 above)
Comment (ranman) [reply cloudformationTemplates/startChatContactAPI/README.md:49]: even in the picture you posted the two script tags are indented once already inside of the code block. you have extra whitespace.

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 11 above)
Comment (ranman) [reply cloudformationTemplates/startChatContactAPI/README.md:65]: do you see how where it says
<script>
then there are two lines before $(document)  ? that's double indenting.

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 3 above)
Comment (yvprashanth) [reply cloudformationTemplates/startChatContactAPI/README.md:15]: I did. The first time I ran it I got a weird error that states i Need a contact flow that's valid
